sorc/build_all.sh

```diff
@@ -228,11 +257,31 @@ while [[ ${builds_started} -lt ${#build_jobs[@]} ]]; do
       fi
    done
 
+   # If requested, check if any build has failed and exit if so
+   if [[ "${_quick_kill}" == "YES" ]]; then
+      check_builds
+      build_stat=$?
+      if [[ ${build_stat} != 0 ]]; then
+         exit "${build_stat}"
+      fi
+   fi
+
    sleep 5s
 done
 
+
 # Wait for all jobs to complete and check return statuses
-while [[ ${#build_jobs[@]} -gt 0 ]]; do
+while [[ "${#build_jobs[@]}" -gt 0 ]]; do
+
+   # If requested, check if any build has failed and exit if so
```

Comment:
I'm not sure we need this in both places. @DavidHuber-NOAA can you take a look at this script's changes?